docs/how-to/llms/custom.md

```diff
@@ -0,0 +1,118 @@
+# How-To: Create Custom LLM
+
+LLM is one of the main components of the db-ally ecosystem. It handles all interactions with the selected Large Language Model. It is used for operations like view selection, IQL generation and natural language response generation, therefore it is essential to be able to integrate with any LLM API you may encounter.
+
+## Implementing a Custom LLM
+
+The `LLM` class is an abstract base class that provides a framework for interacting with a Large Language Model (LLM). To create a custom LLM, you need to create a subclass of `LLM` and implement the required methods and properties.
+
+Here's a step-by-step guide:
+
+### Step 1: Define the subclass
+
+First, define your subclass and specify the type of options it will use:
+
+```python
+from dbally.llms.base import LLM
+from dbally.llms.litellm import LiteLLMOptions
+
+class MyLLM(LLM[LiteLLMOptions]):
+    _options_cls = LiteLLMOptions
+```
+
+In this example we will be using `LiteLLMOptions`, which contain all options supported by most popular LLM APIs. If you need a different interface, see [Customising LLM Options](#customising-llm-options) to learn how to implement it.
+
+### Step 2: Create the custom LLM client
+
+The `_client` property is an abstract method that must be implemented in your subclass. This property should return an instance of `LLMClient` that your LLM will use to interact with the model:
```

Comment:
I think it would be nice to link the API Reference page in every reference like `LLMClient` or `LiteLLMOptions`